src/librustc_middle/ty/mod.rs

```diff
@@ -1046,6 +1046,51 @@ impl<'tcx> Predicate<'tcx> {
     pub fn kind(self) -> &'tcx PredicateKind<'tcx> {
         &self.inner.kind
     }
+
+    /// Returns the inner `PredicateAtom`.
+    ///
+    /// Note that this method panics in case this predicate has unbound variables.
```

Comment:
Is this...true?

Reply:
it was not as this caused some ICE I didn't want to deal with yet, it now is